internal/provider/datasources/deployment_test.go

```diff
@@ -44,6 +44,7 @@ data "prefect_deployment" "test_by_id" {
 
 data "prefect_deployment" "test_by_name" {
   name = prefect_deployment.test.name
+  flow_name = prefect_flow.test.name
```

Comment:
I suspect this was only passing before because there was only one deployment in the test workspace, meaning the existing logic actually worked. This obviously breaks in the real world when there is more than one deployment (of any name).